### ISSUE TITLE
PHP Notice upon plugin deactivation.

### ISSUE THREAD

**Issue by rheinardkorf** [xwp/stream#1228]
# Bug Report

## Expected Behavior

Expecting clean logs upon plugin deactivation.

## Actual Behavior

Getting the following exception when deactivating the plugin on both multisite, single site, and per site deactivation.

```
Exception has occurred.
Warning: WP_Stream\DB::WP_Stream\{closure}(): Argument #2 ($key) must be passed by reference, value given
```

Source: https://github.com/xwp/stream/blob/develop/classes/class-db.php#L60-L64

## System Information

- Stream plugin version: 3.6.1

**Comment by kasparsd** [comment]
I wonder if this is related to PHP8 per #1202.

**Comment by delawski** [comment]
Let's check if this is still an issue.

**Comment by delawski** [comment]
> I wonder if this is related to PHP8 per https://github.com/xwp/stream/issues/1202.

It looks like indeed this was fixed as part of #1202

I think that originally the `array_walk()`'s callback function started to throw exceptions after upgrading to PHP 8. Since the original implementation of the sanitization has been revamped in #1274, `array_walk()` is no longer even used there:

https://github.com/xwp/stream/commit/e83368270a36dbb277f93cc4a60ad2b909173165#diff-cef3783287431afea6d2aae22a061f134820cab4cb758e16152c4e984d4723caL58-L65 

I was not able to reproduce the issue anymore so I'm closing this ticket.
